app/js/referenceSearch.controller.js

```diff
@@ -1,190 +1,193 @@
+            vm.lastPage;
+
+            //Page dividing
+            vm.sliceFrom;
+            vm.sliceTo;
+
+            //Page range
+            vm.viewRangeStart;
+            vm.viewRangeEnd;
+
+            // Method used to apply Rest response to controller variables and apply it on template
+            vm.refreshResponse;
+
+            //Method used to prevent app from querying server with every letter input into search query panel
+            vm.timeoutRefresh;
+
+            //Init method
+            activate();
+
+
+            //ng-show logic
+            function isSearching() {
+                return vm.isUserTyping && vm.query.length > 0
+            }
+
+            function isQueryFieldEmpty() {
+                return vm.query.length == 0 && !vm.isUserTyping;
             }
-            else {
-                $scope.$apply(function () {
-                    vm.references='';
-                });
+
+            function conceptsNotFound() {
+                return vm.query.length > 0 && vm.references.length == 0 && !vm.isUserTyping;
             }
-        };
 
-        //Init method
-        function activate(){
-            if ($routeParams.redirectQuery !== undefined) {
-                vm.query = $routeParams.redirectQuery;
-                refreshResponse();
+            function showResults() {
+                return vm.references.length > 0 && vm.query.length > 0 && !vm.isUserTyping;
             }
-            else {
-                vm.query = '';
+
+            function showEntriesCount() {
+                return !vm.isUserTyping && vm.references.length && !vm.loadingMorePages;
             }
 
-            vm.nextPage = nextPage;
-            vm.prevPage = prevPage;
-            vm.firstPage = firstPage;
-            vm.lastPage = lastPage;
-            vm.sliceFrom = sliceFrom;
-            vm.sliceTo = sliceTo;
-            vm.viewRangeStart = viewRangeStart;
-            vm.viewRangeEnd = viewRangeEnd;
-            vm.refreshResponse = refreshResponse;
-            vm.updateResultNotification = updateResultNotification;
-            vm.timeoutRefresh = timeoutRefresh;
+            function showResultInfoAndNavigationButtons() {
+                return !vm.isUserTyping && vm.references.length > 0;
+            }
 
-            //ng-show logic
-            vm.isSearching = isSearching;
-            vm.isQueryFieldEmpty = isQueryFieldEmpty;
-            vm.conceptsNotFound = conceptsNotFound;
-            vm.showResults = showResults;
-            vm.showEntriesCount = showEntriesCount;
-            vm.showResultInfoAndNavigationButtons = showResultInfoAndNavigationButtons;
-            vm.isNextPagePossible = isNextPagePossible;
-            vm.isPrevPagePossible = isPrevPagePossible;
-        }
-    }]);
\ No newline at end of file
+            function isNextPagePossible() {
+                return vm.viewRangeEnd() >= vm.references.length;
+            }
+
+            function isPrevPagePossible() {
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [Expected '===' and instead saw '=='.](https://www.codacy.com/app/openmrs/openmrs-owa-conceptdictionary/file/2321639288/issues/source?bid=3196969&fileBranchId=3199483#l82)